### ISSUE TITLE
Support Workers using CommonJS syntax

### ISSUE THREAD

**Issue by segevfiner** [webpack/webpack#15961]
## Feature request

**What is the expected behavior?**
For Webpack to support workers using CommonJS syntax when targeting node.

The following works in Node:
```js
const { Worker } = require("worker_threads");
const path = require("path");

const worker = new Worker(path.join(__dirname, "./worker.js"));
```

But is not currently transformed by Webpack.

**What is motivation or use case for adding/changing the behavior?**
https://github.com/pinojs/pino/issues/1465

Essentially looking for a way that pino could support bundling it's workers without needing a specialized plugin for each existing bundler.

P.S. It could be useful for libraries starting workers dynamically to be able to get the URL/filename of a transformed worker module bundle instead of just immediately starting it or receiving a custom class that starts it.

**How should this be implemented in your opinion?**
Recognize that as an additional syntax and transform it appropriately in a similar way the existing worker syntax is transformed.

**Are you willing to work on this yourself?**
yes (But I'm not familiar with Webpack internals, and not sure where to start learning them in the docs).

**Comment by alexander-akait** [comment]
We analyze only static `new Worker(...)` syntax, other things are impossible/hard to implement, sorry, we can't analize `path.join(__dirname, "./worker.js")` and undestand URL to worker, you can write a custom plugin to implement this

**Comment by segevfiner** [comment]
Yeah. I know it currently doesn't do that. This is a feature request to add support for that to the internal `WorkerPlugin`. Assuming it is reasonable to try and match such syntax.

**Comment by alexander-akait** [comment]
Why do not use `new URL(...)`? It will work for commonjs too

**Comment by segevfiner** [comment]
We can I guess. But I think `import.metal.url` won't work in CommonJS.

**Comment by alexander-akait** [comment]
@segevfiner anyway you can create a custom plugin, it is not hard - https://github.com/webpack/webpack/blob/main/lib/dependencies/WorkerPlugin.js

@vankop in theory we can add a hook here for better DX, so it will be easy to use own syntax with the worker plugin

**Comment by vankop** [comment]
@alexander-akait what you mean? maybe something like:
```js
class MyPlugin {
  apply(compiler) {
       compiler.hooks.compilation.tap("my", (compilation, normalModuleFactory) => {
          normalModuleFactory.hooks.parser.get("javascript/auto").tap("my", parser => {
              // detect worker
              const request = "./user-request.js";
              const options = {};
              const range = [1, 10];
              WorkerPlugin.addWorker(compilation, request, options, range);
          });
       });
   }
}
```
this could be very specific I guess..

**Comment by vankop** [comment]
maybe we need api to create dynamic entries? sounds like this is correct one...

**Comment by alexander-akait** [comment]
Yes, I think small and clear API will be great here
